integrations/aws/main.py

```diff
@@ -0,0 +1,232 @@
+import json
+import typing
+
+from fastapi import Response, status
+import fastapi
+from starlette import responses
+from pydantic import BaseModel
+
+from aws.overrides import AWSResourceConfig
+from port_ocean.core.models import Entity
+from port_ocean.context.event import event
+
+from utils.resources import (
+    batch_resources,
+    describe_single_resource,
+    fix_unserializable_date_properties,
+    resync_cloudcontrol,
+)
+from utils.config import get_matching_kinds_from_config
+
+from utils.aws import (
+    _session_manager,
+    describe_accessible_accounts,
+    get_sessions,
+    update_available_access_credentials,
+    validate_request,
+)
+from port_ocean.context.ocean import ocean
+from loguru import logger
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from utils.enums import (
+    ACCOUNT_ID_PROPERTY,
+    KIND_PROPERTY,
+    REGION_PROPERTY,
+    ResourceKindsWithSpecialHandling,
+)
+
+
+@ocean.on_resync()
+async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    await update_available_access_credentials()
+
+    if kind in iter(ResourceKindsWithSpecialHandling):
+        logger.info("Kind already has a specific handling, skipping", kind=kind)
+        return
+    async for batch in resync_cloudcontrol(kind):
+        yield batch
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.ACCOUNT)
+async def resync_account(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    for account in describe_accessible_accounts():
+        yield fix_unserializable_date_properties(account)
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.CLOUDRESOURCE)
+async def resync_generic_cloud_resource(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    resource_kinds = typing.cast(
+        AWSResourceConfig, event.resource_config
+    ).selector.resource_kinds
+    for kind in resource_kinds:
+        async for batch in resync_cloudcontrol(kind):
+            yield batch
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.ACM)
+async def resync_acm(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    async for session in get_sessions():
+        async for batch in batch_resources(
+            kind,
+            session,
+            "acm",
+            "list_certificates",
+            "CertificateSummaryList",
+            "NextToken",
+        ):
+            yield batch
```

Comment:
maybe we s
```suggestion
   gens = []
    async for session in get_sessions():
        gen.append(batch_resources(
            kind,
            session,
            "acm",
            "list_certificates",
            "CertificateSummaryList",
            "NextToken",
        ))
    async for x in merge(gens):
      yield x
```

Reply:
Wouldn't it work exactly the same way as it's working today?